DivideAndConquer/StrassenMatrixMultiplication.java

```diff
@@ -0,0 +1,181 @@
+// Java Program to Implement Strassen Algorithm
+ 
+// Class Strassen matrix multiplication
+public class Strassen {
+ 
+    // Method 1
+    // Function to multiply matrices
+    public int[][] multiply(int[][] A, int[][] B)
+    {
+        int n = A.length;
+
+        int[][] R = new int[n][n];
+ 
+        if (n == 1)
+
+            R[0][0] = A[0][0] * B[0][0];
+
+        else {
+            // Dividing Matrix into parts
+            // by storing sub-parts to variables
+            int[][] A11 = new int[n / 2][n / 2];
+            int[][] A12 = new int[n / 2][n / 2];
+            int[][] A21 = new int[n / 2][n / 2];
+            int[][] A22 = new int[n / 2][n / 2];
+            int[][] B11 = new int[n / 2][n / 2];
+            int[][] B12 = new int[n / 2][n / 2];
+            int[][] B21 = new int[n / 2][n / 2];
+            int[][] B22 = new int[n / 2][n / 2];
+ 
+            // Dividing matrix A into 4 halves
+            split(A, A11, 0, 0);
+            split(A, A12, 0, n / 2);
+            split(A, A21, n / 2, 0);
+            split(A, A22, n / 2, n / 2);
+ 
+            // Dividing matrix B into 4 halves
```

Comment:
```suggestion
            // Dividing matrix B into 4 parts
```
4 halves? I thought there are only 2 of them...